tools/label-check.py

```diff
@@ -0,0 +1,182 @@
+    community_modules = glob.glob(community_modules_glob)
+    core_modules_glob = os.path.join(root_dir, "modules", "*", "*")
+    core_modules = glob.glob(core_modules_glob)
+    return community_modules + core_modules
+
+
+def has_labels_variable(module_path: ModulePath) -> bool:
+    if not module_path.has_vars():
+        return False
+    with open(module_path.vars()) as var_file:
+        return 'variable "labels"' in var_file.read()
+
+
+def check_for_labels_local_block(module_path: ModulePath) -> bool:
+    check_string = LOCALS_TEMPLATE.format(
+        module_label=module_path.name_label())
+    file_to_check = module_path.primary_file()
+    with open(file_to_check) as file:
+        return check_string in file.read()
+
+
+def add_labels_local_block(module_path: ModulePath):
+    file_to_write = module_path.primary_file()
+    with open(file_to_write, 'r') as main_file:
+        insert_at = -1
+        lines = main_file.readlines()
+        for num, line in enumerate(lines):
+            if '*/' in line:
+                insert_at = num + 1
+                break
+        if insert_at < 0:
+            print('Could not find "*/" in {}'.format(file_to_write))
+            sys.exit(1)
+
+    lines.insert(insert_at, LOCALS_TEMPLATE.format(
+        module_label=module_path.name_label()))
+
+    with open(file_to_write, 'w') as main_file:
+        main_file.writelines(lines)
+
+
+def check_label_usage(module_path: ModulePath) -> bool:
+    passed = True
+    with open(module_path.primary_file()) as file:
+        if file.read().count('var.labels') > 1:  # there will be one reference in local block
+            print("{} contains references to var.labels instead of local.labels".format(
+                module_path.primary_file()))
+            passed = False
+
+    if module_path.primary_file() != module_path.outputs() and module_path.has_outputs():
+        with open(module_path.outputs()) as outputs:
+            if outputs.read().count('var.labels') > 0:
+                print("{} contains references to var.labels instead of local.labels".format(
+                    module_path.outputs()))
+                passed = False
+    return passed
+
+
+def check_provider_meta(module_path: ModulePath) -> bool:
+    """This is enforcing that the provider meta name matches the module name"""
+    version_file_path = module_path.versions()
+    module_name = module_path.name()
+    with open(version_file_path) as version_file:
+        s = version_file.read()
+        if s.count('provider_meta "google') == s.count(
+                'blueprints/terraform/hpc-toolkit:{}'.format(module_name)):
+            return True
+        else:
+            print('{} provider meta does not match module name'.format(
+                version_file_path))
+            return False
+
+
+def check_module(module_path: ModulePath) -> bool:
+    passed = True
+    if not has_labels_variable(module_path):
+        return passed
```

Comment:
`return False`

Reply:
I see, but realized I was short circuiting the check for provider meta if labels were not in the module. Logic was updated.